src/main/java/seedu/address/model/project/Project.java

```diff
@@ -0,0 +1,49 @@
+package seedu.address.model.project;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+/**
+ * Represents a Project.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Project {
+
+    // Components of a project
+    private Name name;
+    private Repository repository;
+    private Deadline deadline;
+    private String client;
+    private String issue;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Project(Name name, Repository repository, Deadline deadline, String client, String issue) {
+        requireAllNonNull(name, repository, deadline, client, issue);
```

Comment:
Perhaps some of the parameters should not be mandatory, to be in line with the optional arguments of `project -a`. Overloaded constructors and default values might be useful here.

Reply:
Does that not depend on the parser file? Inputs into the constructor in the parser file could be null for fields that are optional at that point

Reply:
I am not too sure what you mean by the constructor in the parser file; is it possible to specify which class you are referring to? Thanks!

Reply:
For now, I am thinking we can input null for fields that are optional when we call the constructor for project class, where the constructor gets called

Reply:
I have checked against the `Person` class of AB3 - in this case, they do a null check on the optional parameter, `tags` because in the case where no tags are specified by the user, it defaults to an empty set. However, in our case we are not working with an optional collection e.g. ArrayList, Set, etc., we can either specify a default value or set the optional parameters to null if not specified by the user. For the latter solution, there would be no need for null checks for optional parameters, although I personally feel that a special default value would be more appropriate.

Reply:
thanks for checking, yup, do see the latest commit PR for the changes